eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -435,8 +435,19 @@ public synchronized BackupRegistry get() {
             throw new RuntimeException("Failed to initialize DiscoveryClient!", e);
         }
 
-        if (clientConfig.shouldFetchRegistry() && !fetchRegistry(false)) {
-            fetchRegistryFromBackup();
+        if (clientConfig.shouldFetchRegistry()) {
```

Comment:
I think the logic here is off, you want to
if (shouldFetch) {
  fetch
  if (unsuccessful) {
    log at info that the normal fetch failed
    fetch from backup
    if still unsuccessful, log a detailed msg and throw
  }
}

Right now you're fetching and then fetching again from backup on success.

Reply:
Good catch. I revisited this logic and tried to clean it up. The changes should now require that fetching from both the primary and backup registry fail before applying the `shouldEnforceFetchRegistryAtInit` configuration.